ClipboardMonitor/ProcessHelper.cs

```diff
@@ -140,41 +173,43 @@ public static void Uncover()
             // Save the DACL
             SetProcessSecurityDescriptor(hProcess, dacl);
             _isProtected = false;
+
+            hProcess.Dispose();
         }
 
-        private static RawSecurityDescriptor GetProcessSecurityDescriptor(IntPtr processHandle)
+        private static RawSecurityDescriptor GetProcessSecurityDescriptor(SafeHandle processHandle)
         {
-            if (processHandle == IntPtr.Zero)
+            if (processHandle.DangerousGetHandle() == IntPtr.Zero)
             {
                 throw new ArgumentException("The process handle is invalid.", nameof(processHandle));
             }
 
             var psd = Array.Empty<byte>();
             // Call with 0 size to obtain the actual size needed in bufSizeNeeded
-            _ = NativeMethods.GetKernelObjectSecurity(processHandle, DACL_SECURITY_INFORMATION, psd, 0, out var bufSizeNeeded);
+            _ = NativeMethods.GetKernelObjectSecurity(processHandle.DangerousGetHandle(), DACL_SECURITY_INFORMATION, psd, 0, out var bufSizeNeeded);
             if (bufSizeNeeded > short.MaxValue)
             {
                 throw new Win32Exception();
             }
             // Allocate the required bytes and obtain the DACL
-            if (!NativeMethods.GetKernelObjectSecurity(processHandle, DACL_SECURITY_INFORMATION, psd = new byte[bufSizeNeeded], bufSizeNeeded, out _))
-            {
-                throw new Win32Exception();
-            }
+            if (!NativeMethods.GetKernelObjectSecurity(processHandle.DangerousGetHandle(), DACL_SECURITY_INFORMATION, psd = new byte[bufSizeNeeded], bufSizeNeeded, out _))
```

Comment:
I believe you're a bit more vulnerable to GC and finalization than you were before this change. the `processHandle` variable isn't referenced after it is prepared as an argument for this method call. Per CLR rules, the GC *can* collect `processHandle` *while this p/invoke is in progress*, and finalize it, at which point the handle may become invalid, causing a crash or other failure in your application. This is why they call it `DangerousGetHandle`. 😄 
In CsWin32 we mitigate this problem by calling `DangerousAddRef` first, and `DangerousRelease` after. 

This is an example of the many things that CsWin32-generated code takes into account that the contributors to pinvoke.net wiki do _not_ usually consider because it 'usually works', but can lead to instability, or only works on certain architectures, etc. 

Shameless plug: If you were using the `GetKernelObjectSecurity` method generated by CsWin32, you would have been protected from this bug.